### PR TITLE
build: hotfix release v5.24.1

### PR DESCRIPTION
**Pull request by karrui**
## Hotfix release
This PR fixes a bug where sending of verified SMSes would be blocked if the form admin's free SMS quota has been hit. This prevents  the vicious cycle where
1. Automatically disabling verified sms fields only happen after a verification sms is sent, and the admin has exceeded their quota.
2. sms is blocked from being sent, resulting in (1) not happening, resulting in the verified sms field not being disabled, resulting in sms being blocked... and the cycle continues.

By only checking whether the current field to send SMS is verifiable, it solves the above cycle.
- If user has hit the free quota but still has a verifiable sms field in their form, then
  - when public user invokes sms verification, the sms will be sent and server will disable mobile verification on all their free forms
  - when the next user (on the old verifiable form) invokes the sms verification, the same field will now not be verifiable and correctly return an error

### More context:
Release v5.24.0 (#2626) added a check to prevent verification SMSes from being sent out if the user’s free sms limit has exceeded.
This revealed a bug. If a form's admin has exceeded the free sms verification limit prior to the automation of the disabling of the verifiability of the mobile field (in release v5.23.0 (#2566)), then their verified mobile field (that was not automatically disabled) would stop working due to the check.

Currently, If a user’s free sms limit is hit, all verified mobile fields of forms that have not been onboarded (still using the free tier) will automatically have its verifiable status removed (which reverts the field to a normal phone number field). Works as intended when the user hits their limit.

However, this toggling of verifiable state is only done when the admin account exceeds the free limit tier (10k smses).
This means that users who have already exceeded the free tier prior to the sms limiting feature release will not have their verified sms fields toggled off, and since the limit has exceeded, any SMS sending will be rejected by the server. The public respondent will then see an error message that prompts them to refresh (but that would not fix the issue).

### CODE REVIEW

**Comment by karrui** [comment]
## Tests
- [ ] Set the admin's current sms count to 9999 (1 below limit) (can do this by deleting/creating a new user and populating counts)
- [ ] Create an email form with 2 verifiable mobile fields. Send the first sms. Should succeed. Disabled email should be sent to admins and collaborators.
- [ ] Send second sms. Should now return an error since limit is hit. Refresh the form and the sms fields should not be verifiable.
- [ ] With the admin now at limit (10k), manually toggle on a field's isVerifiable status in the database. 
  - [ ] Refresh the form. The sms field should now be verifiable again. Verify sms. Should receive disabled email again. The field should go back to being unverifiable after a refresh.

**Review by tshuli — DISMISSED** [review]
lgtm with minor nit - should we also remove `SmsLimitExceededError` as error type for `sendNewOtp` in `verification.service.ts`?

**Comment by tshuli** [comment]
Tests ok too

**Comment by tshuli** [comment]
forced push to adjust the error typings as mentioned in comment above

File: src/app/modules/verification/verification.service.ts [hunk 1]
```diff
@@ -479,16 +477,15 @@ const sendOtpForField = (
   | InvalidNumberError
   | MailSendError
   | NonVerifiedFieldTypeError
-  | SmsLimitExceededError
   | OtpRequestError
 > => {
-  const { fieldType, _id } = field
+  const { fieldType, _id: fieldId } = field
```
Comment (timotheeg) [inline src/app/modules/verification/verification.service.ts:482]: 👍 reads much better!

**Review by timotheeg — APPROVED** [review]
lgtm 👍

**Review by tshuli — APPROVED** [review]

**Pull request merged**